index.js

```diff
@@ -82,6 +83,11 @@ class Listr {
 		let tasks;
 		if (this._options.concurrent === true) {
 			tasks = Promise.all(this._tasks.map(task => runTask(task, context, errors)));
+		} else if (typeof this._options.concurrent === 'number' && this._options.concurrent > 0) {
+			tasks = pMap(this._tasks, task => {
+				this._checkAll(context);
```

Comment:
@sindresorhus  Ahhh sorry, last commit was a bad paste. Next one sets it straight... :)